axelor-account/src/main/resources/views/NotificationItem.xml

```diff
@@ -5,16 +5,31 @@
 
 	<grid model="com.axelor.apps.account.db.NotificationItem" title="Notification items"
 		name="notification-item-grid">
-		<field name="invoice" readonly="true" />
+		<field name="invoice" onSelect="action-account-attrs-notification-item-invoice-domain" onChange="action-account-record-notification-item-invoice-onchange"/>
 		<field name="amountPaid" />
+		<field name="typeSelect"/>
 	</grid>
 
 	<form model="com.axelor.apps.account.db.NotificationItem" title="Notification item"
-		name="notification-item-form">
+		name="notification-item-form" onNew="action-account-record-notification-item-type-select">
 		<panel>
-			<field name="invoice" readonly="true" />
+			<field name="invoice" onSelect="action-account-attrs-notification-item-invoice-domain" onChange="action-account-record-notification-item-invoice-onchange"/>
 			<field name="amountPaid" />
+			<field name="typeSelect"/>
 		</panel>
 	</form>
 
+	<action-record name="action-account-record-notification-item-type-select" model="com.axelor.apps.account.db.NotificationItem">
+		<field name="typeSelect" expr="1"/>
+	</action-record>
+	
+	<action-record name="action-account-record-notification-item-invoice-onchange" model="com.axelor.apps.account.db.NotificationItem">
+		<field name="amountPaid" expr="eval: invoice?.amountRemaining"/>
+	</action-record>
+	
+	<action-attrs name="action-account-attrs-notification-item-invoice-domain">
+		<attribute name="domain" for="invoice" if="_parent?.subrogationRelease != null" expr="eval: self.id in (select inv.id from SubrogationRelease sr inner join sr.invoiceSet inv WHERE sr.id = ${_parent?.subrogationRelease?.id})"/>
+		<attribute name="domain" for="invoice" if="_parent?.subrogationRelease == null" expr="self.id in (select inv.id from SubrogationRelease sr inner join sr.invoiceSet inv)"/>
```

Comment:
add filter on status and company, filter string I have sent you have just pasted it without modification , check requirements properly